content/json-rpc.ts

```diff
@@ -101,40 +101,86 @@ class NSUser {
 
 class NSItem {
   /**
-   * Quick-search for items in Zotero.
+   * Search for items in Zotero.
    *
-   * @param terms  Terms as a single string as typed into the search box in Zotero, or an array of tuples as typed into the advanced search box in Zotero
+   * Examples
+   *
+   * - search('') or search([]): return every entries
+   * - search('Zotero'): quick search for 'Zotero'
+   * - search([['title', 'contains', 'Zotero']]): search for 'Zotero' in the Title
+   * - search([['library', 'is', 'My Library']]): search for entries in 'My Library'
+   *   (this function try to resolve the string 'My Library' into is own libraryId number)
+   * - search([['ignore_feeds']]): custom action for ignoring the feeds
+   * - search([['ignore_feeds'], ['quicksearch-titleCreatorYear', 'contains', 'Zotero']]): quick search for 'Zotero' ignoring the Feeds
+   * - search([['creator', 'contains', 'Johnny'], ['title', 'contains', 'Zotero']]): search for entries with Creator 'Johnny' AND Title 'Zotero'
+   * - search([['joinMode', 'any'], ['creator', 'contains', 'Johnny'], ['title', 'contains', 'Zotero']]): search for entries with Creator 'Johnny' OR Title 'Zotero'
+   * - search([['blockStart'], ['creator', 'contains', 'Johnny'], ['title', 'contains', 'Zotero'], ['blockEnd'], ['creator', 'contains', 'Smith']]): search for entries with (Creator 'Johnny' OR Title 'Zotero') AND (Creator 'Smith')
+   *
+   * @param terms  Single string as typed into the search box in Zotero (search for Title Creator Year)
+   *               Array of tuples similar as typed into the advanced search box in Zotero
+   *               (https://github.com/zotero/zotero/blob/9971f15e617f19f1bc72f8b24bb00b72d2a4736f/chrome/content/zotero/xpcom/data/searchConditions.js#L72-L610)
    */
-  public async search(terms: string | [string, string, string][], library?: string | number) {
+  public async search(terms: string
+  | ([string] | [string, string] | [string, string, string | number] | [string, string, string | number, boolean])[]) {
```

Comment:
Should we still support passing the library ID as the second param? I think it's still needed for basic string searches.